src/core/room.ts

```diff
@@ -118,8 +119,16 @@ export class DefaultRoom implements Room {
    * @param realtime An instance of the Ably Realtime client.
    * @param chatApi An instance of the ChatApi.
    * @param logger An instance of the Logger.
+   * @param initAfter The room will wait for this promise to finish before initializing
    */
-  constructor(roomId: string, options: RoomOptions, realtime: Ably.Realtime, chatApi: ChatApi, logger: Logger) {
+  constructor(
```

Comment:
This is quite a bit of code in the constructor, is it worth pulling some of it out into functions to reduce the cognitive complexity a bit? :)

Reply:
I've refactored some bits. It still looks hard to read but moved lots of logic outside the constructor.